src/test/java/org/sagebionetworks/bridge/participantroster/DownloadParticipantRosterWorkerProcessorTest.java

```diff
@@ -0,0 +1,371 @@
+
+    }
+
+    @Test
+    public void studyCoordinatorWithoutOrgId() throws Exception {
+        StudyParticipant participant = new StudyParticipant();
+        participant.setEmail("example@example.org");
+        participant.setEmailVerified(true);
+        participant.addRolesItem(Role.STUDY_COORDINATOR);
+
+        testParticipantRequirements(participant, makeValidRequestNode());
+    }
+
+    @Test
+    public void studyIdNotSponsoredByOrg() throws Exception {
+        StudyParticipant participant = new StudyParticipant();
+        participant.setEmail("example@example.org");
+        participant.setEmailVerified(true);
+        participant.addRolesItem(Role.RESEARCHER);
+        participant.setOrgMembership(ORG_ID);
+
+        ObjectNode requestNode = makeValidRequestNode();
+        requestNode.put("studyId", STUDY_ID);
+
+        testParticipantRequirements(participant, requestNode);
+    }
+
+    @Test
+    public void getCsvHeaders() throws IllegalAccessException {
+        String[] headers = processor.getCsvHeaders();
+        assertStudyParticipantHeaders(headers);
+    }
+
+    @Test
+    public void getStudyParticipantArray() throws IllegalAccessException {
+        StudyParticipant studyParticipant = makeStudyParticipant(1);
+
+        String[] headers = processor.getCsvHeaders();
+        assertStudyParticipantHeaders(headers);
+
+        String[] studyParticipantArrayArray = processor.getStudyParticipantArray(studyParticipant, headers);
+        assertEquals(studyParticipantArrayArray.length, 22);
+
+        assertEquals("test-first-name1", studyParticipantArrayArray[0]);
+        assertEquals("test-last-name", studyParticipantArrayArray[1]);
+        assertEquals("test-id-12345", studyParticipantArrayArray[2]);
+        assertEquals("true", studyParticipantArrayArray[3]);
+        assertEquals("", studyParticipantArrayArray[4]);
+        assertEquals("", studyParticipantArrayArray[6]);
+        assertEquals("", studyParticipantArrayArray[7]);
+        assertEquals("", studyParticipantArrayArray[8]);
+        assertEquals("", studyParticipantArrayArray[9]);
+        assertEquals("", studyParticipantArrayArray[10]);
+        assertEquals("", studyParticipantArrayArray[11]);
+        assertEquals("", studyParticipantArrayArray[12]);
+        assertEquals("", studyParticipantArrayArray[13]);
+        assertEquals("", studyParticipantArrayArray[14]);
+        assertEquals("{\"externalId1\":\"test-id-1\",\"externalId2\":\"test-id-2\",\"externalId3\":\"test-id-3\"}", studyParticipantArrayArray[15]);
+        assertEquals("", studyParticipantArrayArray[16]);
+        assertEquals("test@test.test", studyParticipantArrayArray[17]);
+        assertEquals("", studyParticipantArrayArray[18]);
+        assertEquals("", studyParticipantArrayArray[19]);
+        assertEquals("", studyParticipantArrayArray[20]);
+        assertEquals("", studyParticipantArrayArray[21]);
+    }
+
+    @Test
+    public void process() throws Exception {
```

Comment:
@DwayneJengSage Do you know if there's a test studyId that I can provide that will be considered sponsored by the given test org id? Otherwise, it will fail because of the sponsorship requirement.

Reply:
It comes from getSponsoredStudies().

Reply:
Makes so much sense. Thank you.